scripts/ci-e2e.sh

```diff
@@ -31,7 +31,7 @@ source "${REPO_ROOT}/hack/ensure-kustomize.sh"
 # Configure provider images generation;
 # please ensure the generated image name matches image names used in the E2E_CONF_FILE
 export REGISTRY=gcr.io/k8s-staging-cluster-api
-export TAG=ci
+export TAG=dev
```

Comment:
from @vincepri  https://github.com/kubernetes-sigs/cluster-api/pull/3639#discussion_r488747505

> Why change the tag for the CI?

In order to simplify things, we are now going to maintain a single E2E configuration, instead of maintaining one for CI and one for DEV.
Using `TAG=dev` also in CI is the simplest option because it is the one used by default by all the `make docker-build` targets